src/mame/brother/ax145.cpp

```diff
@@ -0,0 +1,274 @@
+// license:BSD-3-Clause
+// copyright-holders:Bartman/Abyss
+
+#include "emu.h"
+#include "emupal.h"
+#include "screen.h"
+#include "speaker.h"
+#include "machine/timer.h"
+#include "cpu/z180/z180.h"
+#include "debug/debugcpu.h"
+#include "sound/beep.h"
+#include "video/hd44780.h"
+
+// 240x18x4 = 960x72
+// -log -debug -window -intscalex 4 -intscaley 4 -resolution 960x72 ax145
+
+//////////////////////////////////////////////////////////////////////////
+// AX-145
+//////////////////////////////////////////////////////////////////////////
+
+/***************************************************************************
+
+Brother AX-145
+198?
+
+Hardware:
+
+Main PCB B482360-1
+========
+
+#7 Hitachi
+HD64180RP6
+6 MHz, 80xx Peripherals
+
+#6 Texas Instruments
+LM393P
+Dual differential comparator, commercial grade
+
+#5 not populated
+µPD23C2001
+
+#2 NEC
+D43256AC-10L
+NEC
+Static CMOS RAM 32,768 x 8-Bit
+
+#3 NEC
+D23C4001EC-172
+UA2849-A
+4MBit Mask ROM for Dictionary
+
+#4 NEC
+D23C1001EC-522
+UA4474-C
+1MBit Mask ROM
+
+#1 Texas Instruments
+SN74HC00N
+Quad 2-Input Positive-NAND Gates
+
+#8 NEC
+D65013-A73
+UA3001-A
+CMOS Gate Array
+
+LCD PCB connected via 12-pin ribbon cable
+========
+Hitachi
+HD44780S-B02
+U17978-A
+Dot Matrix Liquid Crystal Display Controller/Driver
+Custom Font
+(driven in 4-bit mode, R/W connected)
+
+2x Hitachi
+HD66100F
+LCD Driver with 80-Channel Outputs
+
+LCD: 40 characters x 2 lines
+
+// Status:
+// doesn't go further than "SCHREIBWERK ÜBERPRÜFEN" (check printer)
+// needs european font for HD44780
+
+***************************************************************************/
+
+// hw_config:
+// - & 0x1f:
+//    6            => deutsch
+//    12           => francais/nederlands
+//    13           => nederlands
+//    16, 17       => francais/deutsch
+//    8, 9, 14, 21 => espanol
+
+class ax145_state : public driver_device
+{
+public:
+	ax145_state(const machine_config &mconfig, device_type type, const char *tag)
+		: driver_device(mconfig, type, tag),
+		maincpu(*this, "maincpu"),
+		lcdc(*this, "hd44780"),
+		ram(*this, "ram", 0x8000, ENDIANNESS_LITTLE),
+		rom(*this, "maincpu")
+	{ }
+
+	void ax145(machine_config& config);
+
+private:
+	// devices
+	required_device<hd64180rp_device> maincpu;
+	required_device<hd44780_device> lcdc;
+	memory_share_creator<uint8_t> ram;
+	required_region_ptr<uint8_t> rom;
+
+	// valid values (bei IO3000=0x0b,0x07) (read_config @ 0x14c87): 0 = german => 0, 1 = german => 1, 2 = espanol => 2, 4 (gehäusedeckel offen) => 3, 8 = francais => 4, 16 = german => 5
+	static constexpr uint8_t id = 1;
+
+	// config switch
+	uint8_t io_3000{};
+
+	// bit 2: R/~W (Read / Not Write)
+	// bit 1: RS (Register Select)
+	// bit 0: E (Enable)
+	uint8_t lcd_signal{};
+
+	// driver_device overrides
+	void machine_start() override;
+	void machine_reset() override;
+	void video_start() override;
```

Comment:
Please make these `protected` like they are in the base class they’re inherited from.